src/main/webapp/assets/modules/technical-detail/technical-detail.con.js

```diff
@@ -1,3 +1,23 @@
-techlooper.controller("technicalDetailController", function() {
+techlooper.controller("technicalDetailController", function (utils, connectionFactory, $routeParams,
+                                                             technicalDetailService, $scope, $timeout, jsonValue, termService) {
+  var term = $routeParams.term;
 
+  // TODO: write a blog about dom manipulation with angularjs
+  $scope.showCircle = function(skill) {
+    technicalDetailService.showSkillsList(skill);
+    return true;
+  }
+
+  $scope.companyUrl = function(company) {//java-fpt+at-it-software-i35-en
+    return sprintf("http://vietnamworks.com/%s-%s+at-it-software-i35-en", term.toLowerCase(), company.name);
+  }
+
+  connectionFactory.termStatisticInOneYear({term: term})
+    .success(function (data, status, headers, config) {
+      $scope.termStatistic = termService.toViewTerm(data);
+      console.log($scope.termStatistic);
+      technicalDetailService.trendSkills($scope.termStatistic);
+    })
+    .error(function (data, status, headers, config) {
```

Comment:
'config' is defined but never used.<br>'headers' is defined but never used.<br>'status' is defined but never used.<br>'data' is defined but never used.